File: baekjoon/Level/Silver/11055. 가장 큰 증가하는 부분 수열/가장 큰 증가하는 부분 수열.swift
let N = Int(readLine()!)!
let nums = readLine()!.split(separator: " ").map{ Int($0)! }

var dp = nums
for i in 0..<N {
    for j in 0..<i {
        if nums[j] < nums[i] {
            dp[i] = max(dp[i], dp[j]+nums[i])
        }
    }
}

print(dp.max()!)

File: baekjoon/Level/Silver/11055. 가장 큰 증가하는 부분 수열/가장 큰 증가하는 부분 수열.py
N = int(input())
nums = list(map(int, input().split(" ")))

dp = [] + nums
for i in range(N):
    for j in range(0, i):
        if nums[j] < nums[i]:
            dp[i] = max(dp[i], dp[j]+nums[i])

print(max(dp))